com.unity.formats.fbx/Editor/FbxExporter.cs

```diff
@@ -3136,14 +3136,19 @@ int objectCount
         /// components and the animation clips. Also, the first animation to export, if any.
         /// </summary>
         /// <returns>The animation only hierarchy count.</returns>
-        /// <param name="exportSet">GameObject hierarchies selected for export.</param>
         /// <param name="hierarchyToExportData">Map from GameObject hierarchy to animation export data.</param>
         [SecurityPermission(SecurityAction.LinkDemand)]
         internal int GetAnimOnlyHierarchyCount(Dictionary<GameObject, IExportData> hierarchyToExportData)
         {
             // including any parents of animated objects that are exported
             var completeExpSet = new HashSet<GameObject>();
-            foreach (var data in hierarchyToExportData.Values) {
+            foreach (var item in hierarchyToExportData) {
+                var data = item.Value;
+                if(data == null || data.Objects.Count <= 0)
+                {
+                    Debug.LogWarningFormat("No animation data to export for objects in hierarchy: {0}", item.Key);
```

Comment:
Do we need this warning?

It feels wrong on two levels:
- if I select a bunch of objects and some of them have anim but others don't, I shouldn't need to care about those that don't.
- the caller should be in charge of determining what to do about there's no animation data to export. We should just report `0` and be happy.

Reply:
yeah you're right maybe not necessary